.github/workflows/release.yml

```diff
@@ -0,0 +1,46 @@
+name: Release
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create Release Artifacts
+        run: make release
+        env:
+          IMAGE_VERSION: ${{ github.ref_name }}
+      - name: Upload Release Artifacts
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            ./release/helm-chart-${{ github.ref_name }}.tgz
+            ./release/linode-blockstorage-csi-driver-${{ github.ref_name }}.yaml
+      - name: Docker Meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            linode/linode-blockstorage-csi-driver
+          tags: |
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=semver,pattern={{raw}},value=${{ github.ref_name }}
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and Push to Docker Hub
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: "./app/linode/Dockerfile"
```

Comment:
One big downside of this is that this is _only_ exercised during pushes of tags - changes to the Dockerfile, if broken, would go unnoticed.

The big upside of the previous implementation was that line `no_push: ${{ github.event_name != 'release' }}` (which I believe has an alternative in `docker/build-push-action`). This allowed us to run this on each commit without sending this image anywhere.

Reply:
how about we add the docker build to the ci process? i prefer keeping the ci/release actions separate so you don't accidently do something and miss a conditional gate or something.

Reply:
https://github.com/linode/linode-blockstorage-csi-driver/actions/runs/6800642763/job/18489660804 how do you like this?

Reply:
1. Why is this resolved when I didn't have a say in this?
2. This is still not quite what I had in mind (though better than before), because you're using two different build methods - one is a make target that never gets triggered during a release, the other is an Action that does the build and push.

Reply:
1. clicking things too fast, it wasn't supposed to be resolved, sorry about that.
2. prefer to have the same docker actions and steps in the ci job with no push? let me push those changes and let's see how we like that

Reply:
updated and added a parallel docker build. it's a bit dupe effort with pulling modules and doing `go build` go build twice... so i split into two steps which will hopefully complete at the same time.

weirdly shuffling these things around I created a bug which by removing the `push` arg which has now been fixed on both jobs